src/mono/System.Private.CoreLib/src/System/Exception.Mono.cs

```diff
@@ -66,34 +68,46 @@ internal DispatchState CaptureDispatchState()
         {
             MonoStackFrame[]? stackFrames;
 
-            if (_traceIPs != null)
+            // Make sure foreignExceptionFrames does not change at this point.
+            // Otherwise, the Array.Copy into combinedStackFrames can fail due to size differences
+            //
+            // See https://github.com/dotnet/runtime/issues/70081
+            lock(frameLock)
```

Comment:
Can this be done without a lock by storing `foreignExceptionsFrames` into a local variable before this line https://github.com/dotnet/runtime/blob/5fe7d36573257cb9cde5beea60602610c8361be2/src/mono/System.Private.CoreLib/src/System/Exception.Mono.cs#L83 and using the local variable there?